SWIPs/swip-honey_to_money.md

```diff
@@ -9,29 +9,29 @@ Created: 31-07-2019
 ---
 
 ## Simple summary 
-Swarm needs a mechanism to allow fluctuation of the exchange rate between honey (Swarms internal accounting unit) and the currency used to pay. We propose an on-chain price oracle that returns an exchange rate + validUntil tuple.  
+Swarm needs a mechanism to allow fluctuation of the exchange rate between honey (Swarms internal accounting unit) and the currency used to pay. We propose an on-chain price oracle that returns an `(exchange rate,validUntil)` tuple.  
 
 ## Abstract 
 The ability to update prices is required in order to:
 * Facilitate experimentation with the absolute price of honey (e.g. start at a low price and gradually increase);
-* Off-set fluctuations of the currency used to pay;
-* Lower the price of Swarm in the long-run as the costs of operating a Swarm node decreases.
+* Offset fluctuations of the currency used to pay;
+* Lower the price of Swarm in the long run as the costs of operating a Swarm node decreases.
 
 We propose an on-chain price oracle that returns the current exchange rate when queried. The price-oracle will be initially managed by Swarm developers and stakeholders.
```

Comment:
This of course opens up the big issue of governance.

We don't mention this anywhere. Although it says "initially", I wonder if we should loose some words about governance. And even if it is only *initially* managed the devs. we should think (and in my opinion add here) how this is transparently handled to the community. Will we advertise price changes? How? Is it automatic? Do users need to even know?

Reply:
I like it that you bring this up. The oracle model brings a centralized component in an otherwise decentralized (peer-to-peer) network; we are forcing the nodes to use our oracle if they can't come to an agreement of other Oracles (see: [SWIP-multipe_payment_processing](https://github.com/Eknir/SWIPs/blob/master/SWIPs/swip-multiple_payment_processing_support.md), which, in my opinion, is something very serious and should be considered as such. 

To answer directly your point on governance: I agree that it should be at least mentioned here as a pointer for the future; I am not sure about the need to define proper governance at this point. We should get the feedback from the rest of Swarm here.

Reply:
As someone who is active in DAO research, I think it is premature to decide on the mechanism to use. The field is too new and too immature. There is a lot of really interesting research and experimentation going on and I am confident we will find better methods to use in future.

I do object however to the notion that this _introduces_ a centralised component. It is more accurate to say that it _exposes_ and _increases visibility_ of the centralised power of the swarm developers while simultaneously paving the way for its dissipation.
If prices are hard coded, then our coordinating function is hidden. If we publish it in an on-chain oracle, it is visible and explicit and furthermore, it allows others to run their own oracles and convince swarm nodes to start using those with only a configuration change (not a code change). I think this is an improvement.

Reply:
I do not mean to say that we need the *definitive* and "final" way of how governance is going to be done. 

But I do think that it should at least be stated *how we start*.

Who will be responsible to set the prices? Is it the incentives track? Is it the research track? Is it the Swarm team? How? Will we be doing weekly meetings? Daily? 

Mind you I am not talking about the initial development phase, where there will be a lot of experimentation. It will be object of the next weeks and months to deploy test networks and observe and *start* to fine tune all prices; I would assume that for this period we will unilaterally and without much ado manipulate prices at will.

But we can't IMHO bring in the notion of oracles without thinking how we manage these things in a beta network or mainnet 1.0. Not 2.0, not 3.0, but 1.0. We cannot keep kicking the cans down the read in respect to incentives.